debezium-storage/debezium-storage-s3/src/main/java/io/debezium/storage/s3/history/S3SchemaHistory.java

```diff
@@ -0,0 +1,291 @@
+            .withWidth(Width.LONG)
+            .withDefault("")
+            .withImportance(Importance.HIGH);
+
+    public static final Field SECRET_ACCESS_KEY = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + SECRET_ACCESS_KEY_CONFIG)
+            .withDisplayName("S3 secret access key")
+            .withType(Type.PASSWORD)
+            .withWidth(Width.LONG)
+            .withDefault("")
+            .withImportance(Importance.HIGH);
+
+    public static final Field REGION = Field.create(REGION_CONFIG)
+            .withDisplayName("S3 region")
+            .withWidth(Width.LONG)
+            .withType(Type.STRING)
+            .withImportance(Importance.MEDIUM);
+
+    public static final Field BUCKET = Field.create(BUCKET_CONFIG)
+            .withDisplayName("S3 object")
+            .withType(Type.STRING)
+            .withImportance(Importance.HIGH);
+
+    public static final Field ENDPOINT = Field.create(ENDPOINT_CONFIG)
+            .withDefault("S3 endpoint")
+            .withType(Type.STRING)
+            .withImportance(Importance.LOW);
+
+    public static final Field.Set ALL_FIELDS = Field.setOf(ACCESS_KEY_ID, SECRET_ACCESS_KEY, REGION, BUCKET, ENDPOINT);
+
+    private final AtomicBoolean running = new AtomicBoolean();
+    private final FunctionalReadWriteLock lock = FunctionalReadWriteLock.reentrant();
+    private final DocumentWriter writer = DocumentWriter.defaultWriter();
+    private final DocumentReader reader = DocumentReader.defaultReader();
+    private final String objectName = String.format("db-history-%s.log", Thread.currentThread().getName());
+
+    private String bucket = null;
+    private Region region = null;
+    private URI endpoint = null;
+    private AwsCredentialsProvider credentialsProvider = null;
+
+    private volatile S3Client client = null;
+    private volatile List<HistoryRecord> records = new ArrayList<>();
+
+    @Override
+    public void configure(Configuration config, HistoryRecordComparator comparator, SchemaHistoryListener listener, boolean useCatalogBeforeSchema) {
+        super.configure(config, comparator, listener, useCatalogBeforeSchema);
+        if (!config.validateAndRecord(ALL_FIELDS, LOGGER::error)) {
+            throw new SchemaHistoryException("Error configuring an instance of " + getClass().getSimpleName() + "; check the logs for details");
+        }
+
+        bucket = config.getString(BUCKET);
+        region = Region.of(config.getString(REGION));
+        endpoint = URI.create(config.getString(ENDPOINT));
+
+        if (config.getString(ACCESS_KEY_ID).isEmpty() && config.getString(SECRET_ACCESS_KEY).isEmpty()) {
+            credentialsProvider = AnonymousCredentialsProvider.create();
+        }
+        else {
+            AwsCredentials credentials = AwsBasicCredentials.create(config.getString(ACCESS_KEY_ID), config.getString(SECRET_ACCESS_KEY));
+            credentialsProvider = StaticCredentialsProvider.create(credentials);
+        }
+
```

Comment:
is it possible to use [DefaultCredentialsProvider](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html#credentials-chain) or allow usage of other credential providers like InstanceProfileCredentialsProvider ?

it will enable using issuance profile credentials. for security user might not want to generate access,secret key

Reply:
Reading the docs this sounds like a good idea. I can even imagine keeping those auth related config options as optional ones. If present then they'll be converted and set as system properties so they'll be picked-up by `DefaultCredentialsProvider` too. And all other ways how to configure authentication will still be available.

Reply:
I just changed from `AnonymousCredentialsProvider` to `DefaultCredentialsProvider`. In the case that `access_key` and `private_key` the credentials would be `StaticCredentialsProvider`.
At the first thought, my Idea was to provide a field such as `s3.credentialsProvider.class` to explicitly get the `CredentialsProvider` from user but because of the AWS SDK Builder pattern, I didn't.